kessel/auth/auth.go

```diff
@@ -0,0 +1,150 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/zitadel/oidc/v3/pkg/client"
+)
+
+const expirationWindow = 300  // 5 minutes in second
+const defaultExpiresIn = 3600 // 1 hour in seconds
+
+type OIDCDiscoveryMetadata struct {
+	TokenEndpoint string
+}
+
+type RefreshTokenResponse struct {
+	AccessToken string
+	ExpiresAt   time.Time
+}
+
+type OAuth2ClientCredentials struct {
```

Comment:
Should we add a properties section to the OAuth2ClientCredentials class in the spec to specify what attributes get initialized?